src/helperFunctions/convertDatetoYMD.ts

```diff
@@ -0,0 +1,30 @@
+export default function convertDatetoYMD(timestamp: string) {
+  const today = new Date();
+  const idleSincedate = new Date(timestamp);
+  const dateArray = idleSincedate.toISOString().split('-');
+  const year = today.getFullYear();
+  const month = today.getMonth() + 1;
+  const day = today.getDate();
+  const yy = parseInt(dateArray[0]);
+  const mm = parseInt(dateArray[1]);
+  const dd = parseInt(dateArray[2]);
+  let years, months;
+  // months
+  months = month - mm;
+  if (day < dd) {
+    months = months - 1;
+  }
+  // years
+  years = year - yy;
+  if (month * 100 + day < mm * 100 + dd) {
+    years = years - 1;
+    months = months + 12;
+  }
+  // days
+  const days = Math.floor(
+    (today.getTime() - new Date(yy + years, mm + months - 1, dd).getTime()) /
+      (24 * 60 * 60 * 1000)
+  );
+
+  return `${years} years , ${months} months , ${days} days`;
```

Comment:
instead of passing the parameter `timestamp` we can call `idleSince`.  We can calculate the `ms` difference between the 2 timestamps and divide by 86400000 to get the no. of days then from the total day's count you can figure out the year-month-days. If that sounds long then you can go with this approach as well. Both do the same thing. 
```js
const today = new Date().getTime()
const diffInMs = today - idleSince
let noOfdaysIdleSince = diffInMs / 86400000; //86400000 ms in a day
```

Reply:
Ah, yeah I tried doing that, got too complicated for no reason so, avoided it